File: questao_04/questao_03.java
abstract class Animal {
    public abstract void som();
}

class Cachorro extends Animal {
    @Override
    public void som() {
        System.out.println("O cachorro faz: Au Au!");
    }
}

class Gato extends Animal {
    @Override
    public void som() {
        System.out.println("O gato faz: Meow!");
    }
}

public class questao_03 {
    public static void main(String[] args) {
        Animal meuCachorro = new Cachorro();
        Animal meuGato = new Gato();

        meuCachorro.som();
        meuGato.som();
    }
}

File: questao_04/questao_03.go
package main

import (
	"fmt"
)

type Animal interface {
	Som()
}

type Cachorro struct{}

func (c Cachorro) Som() {
	fmt.Println("O cachorro faz: Au Au!")
}

type Gato struct{}

func (g Gato) Som() {
	fmt.Println("O gato faz: Meow!")
}

func main() {
	animais := []Animal{Cachorro{}, Gato{}}

	for _, animal := range animais {
		animal.Som()
	}
}
